torch/testing/_internal/distributed/multi_threaded_pg.py

```diff
@@ -135,7 +139,7 @@ def _end_coll(cls, collective):
             if cls._cur_coll == collective:
                 cls._cur_coll = None
 
-    def allgather(self, output_tensors, input_tensor, options):
+    def allgather(self, output_tensors, input_tensor, opts=AllgatherOptions()):
```

Comment:
`opts` is not used in this function. Why do we need to add it here?

Reply:
process group API requires `opts` for every collective, and it need to have a default value. The c10d API does not require it, it will create `opts` under the hood